### DISCUSSION TITLE
Question: DQ over AS - How to retrieve new/renamed measures?

### DISCUSSION THREAD

**Discussion by GMedlin** [TabularEditor/TabularEditor3#1129] [Ideas]
We have a composite model using DQ over AS to a Power BI Dataset. In this dataset, we added a new measure. The only way we've seen to get that new measure to show up in the composite model is to open it with Power BI Desktop and click Refresh. This causes certain difficulties, especially for models we want to manage with Tabular Editor.

Is there a method in Tabular Editor to update the composite model with any new measures that've been added to the source dataset? Sort of like how Schema Update works to detect DQ table changes, but for measures?

Related to DQ over AS - we may put in a ticket with Microsoft on a scenario where a client is using a Power BI deployment pipeline to promote changes to their composite model (with DQ to PBI dataset). We add a new measure to our "gold source" dataset, which that composite model uses, but there's no way within the service we've seen to have it detect and bring in the new measure to the composite model. Having them use Tabular Editor to do this, assuming it were possible, wouldn't be a great option for the client's skillset and wouldn't want to disable downloading the dataset.

It's been a preview feature for a long time and we try to dissuade folks from using it for "production" datasets, but its apparent ease-of-use up front is what attracts them to it. By the time they understand all the limitations, they're a bit stuck.

**Comment by rem-bou** [comment]
Hi @GMedlin 

We currently have similar issue where we need to use DQ over AS. I've reach out to TE team about it and they told me they aren't supporting this feature yet.

I've just uploaded a [script on my repo](https://github.com/rem-bou/TabularEditor-Scripts/blob/main/Advanced/One-Click%20Macros/Create-Update%20DQ%20over%20AS%20model%20connection.csx) I've developed to do exactly what you describe.

We haven't put into a "production" process but if it can help you, happy to share it.
Let me know if you see any issues with it.

**Comment by GMedlin** [comment]
@rem-bou - Cool script, thanks for sharing!
